test/route.test.js

```diff
@@ -217,6 +217,59 @@ test('same route definition object on multiple prefixes', async t => {
   await fastify.ready()
 })
 
+test('mutating the route definition in onRoute hooks shouldnt interfere with afterRouteAdded() being called twice', t => {
```

Comment:
Might be good to refer the issue from fastify compress in the code:
https://github.com/fastify/fastify-compress/pull/141